client/components/searchbar.js

```diff
@@ -1,13 +1,36 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import history from '../history'
 
-export const SearchBar = props => {
-  return (
-    <span className="search-bar">
-      <form>
-        <input type="text" name="searchText" id="searchText" />
-        <button type="submit">Search</button>
-      </form>
-    </span>
-  )
+export default class SearchBar extends React.Component {
+  constructor(){
+    super()
+    this.state = {
+      term: ""
+    }
+    this.onFormSubmit = this.onFormSubmit.bind(this)
+    this.onFormChange = this.onFormChange.bind(this)
+  }
+  onFormSubmit(event) {
+    event.preventDefault()
+    let poop = this.state.term
+    console.log(poop)
+    history.push(`/products/search/${poop}/page/0`)
+  }
+  onFormChange(event) {
+    this.setState({term: event.target.value})
+  }
+
+  render() {
+    return (
+      <span className="search-bar">
+        <form onSubmit={this.onFormSubmit}>
+          <input type="text" name="searchText" id="searchText" onChange={this.onFormChange}/>
+          <button type="submit">Search</button>
+        </form>
+      </span>
+    )
```

Comment:
This all looks pretty good but when I spin up the server with this code I can't search-- I get a not found for the search result route, and searching redirects to the homepage. I wonder if it's a problem with the routes?

I think it's pretty typical to pass in search terms as a query (after the ?=) at the end of a URL rather than as a URL param. But that's probably not a huge deal